File: 1297-maximum-number-of-balloons/maximum-number-of-balloons.cpp
class Solution {
public:
    int maxNumberOfBalloons(string t){
        int n=t.size();

        map<char, int> mp;
        for(char ch: t){
            if(ch == 'b' || ch == 'a' || ch == 'l' || ch == 'o' || ch == 'n'){
                mp[ch]++;
            }
        }

        if(mp.size()<5)
            return 0;

        int res=INT_MAX;
        for(char ch: t){
            if(ch == 'b' || ch == 'a' || ch == 'n'){
                res=min(res, mp[ch]);
            }
            else if(ch == 'l' || ch == 'o'){   
                res=min(res, mp[ch]/2);
            }
        }

        return res;

    }
};

File: 1297-maximum-number-of-balloons/maximum-number-of-balloons.java
class Solution{
    public int maxNumberOfBalloons(String t){
        int n=t.length();

        HashMap<Character, Integer> mp=new HashMap<>();
        for(char ch: t.toCharArray()){
            if(ch == 'b' || ch == 'a' || ch == 'l' || ch == 'o' || ch == 'n'){
                mp.put(ch, mp.getOrDefault(ch, 0)+1);
            }
        }

        if(mp.size()<5)
            return 0;

        int res=Integer.MAX_VALUE;
        for(char ch: t.toCharArray()){
            if(ch == 'b' || ch == 'a' || ch == 'n'){
                res=Math.min(res, mp.get(ch));
            }
            else if(ch == 'l' || ch == 'o'){   
                res=Math.min(res, mp.get(ch)/2);
            }
        }

        return res;

    }
}